metadata-ingestion/src/datahub/ingestion/source/glue_schema_registry.py

```diff
@@ -0,0 +1,404 @@
+
+    def get_schema_str_replace_confluent_ref_avro(
+            self, schema: Schema, schema_seen: Optional[set] = None
+    ) -> str:
+        if not schema.references:
+            return self._compact_schema(schema.schema_str)
+
+        if schema_seen is None:
+            schema_seen = set()
+        schema_str = self._compact_schema(schema.schema_str)
+        for schema_ref in schema.references:
+            ref_subject = schema_ref["subject"]
+            if ref_subject in schema_seen:
+                continue
+
+            if ref_subject not in self.known_schema_registry_subjects:
+                logger.warning(
+                    f"{ref_subject} is not present in the list of registered subjects with schema registry!"
+                )
+
+            reference_schema = self.glue_client.get_latest_version(
+                subject_name=ref_subject,
+            )
+            schema_seen.add(ref_subject)
+            logger.debug(
+                f"ref for {ref_subject} is {reference_schema.schema.schema_str}"
+            )
+            # Replace only external type references with the reference schema recursively.
+            # NOTE: The type pattern is dependent on _compact_schema.
+            avro_type_kwd = '"type"'
+            ref_name = schema_ref["name"]
+            # Try by name first
+            pattern_to_replace = f'{avro_type_kwd}:"{ref_name}"'
+            if pattern_to_replace not in schema_str:
+                # Try by subject
+                pattern_to_replace = f'{avro_type_kwd}:"{ref_subject}"'
+                if pattern_to_replace not in schema_str:
+                    logger.warning(
+                        f"Not match for external schema type: {{name:{ref_name}, subject:{ref_subject}}} in schema:{schema_str}"
+                    )
+                else:
+                    logger.debug(
+                        f"External schema matches by subject, {pattern_to_replace}"
+                    )
+            else:
+                logger.debug(f"External schema matches by name, {pattern_to_replace}")
+            schema_str = schema_str.replace(
+                pattern_to_replace,
+                f"{avro_type_kwd}:{self.get_schema_str_replace_confluent_ref_avro(reference_schema.schema, schema_seen)}",
+            )
+        return schema_str
+
+    def get_schemas_from_confluent_ref_protobuf(
+            self, schema: Schema, schema_seen: Optional[Set[str]] = None
+    ) -> List[ProtobufSchema]:
+        all_schemas: List[ProtobufSchema] = []
+
+        if schema_seen is None:
+            schema_seen = set()
+
+        schema_ref: SchemaReference
+        for schema_ref in schema.references:
+            ref_subject: str = schema_ref["subject"]
+            if ref_subject in schema_seen:
+                continue
+            reference_schema: RegisteredSchema = (
+                self.schema_registry_client.get_latest_version(ref_subject)
```

Comment:
What is the different between `glue_client.get_schema_version` vs  `schema_registry_client.get_latest_version` and `glue_client.get_latest_version`. Can we use same method at all places ?